src/control/server/instance_storage_rpc.go

```diff
@@ -206,11 +210,22 @@ func scanEngineBdevsOverDrpc(ctx context.Context, engine Engine, pbReq *ctlpb.Sc
 			c.SmdDevices = nil
 			c.HealthStats = nil
 			seenCtrlrs[addr] = c
-			pbResp.Ctrlrs = append(pbResp.Ctrlrs, c)
 		}
 
 		c := seenCtrlrs[addr]
 
+		// Only minimal info provided in standard scan to enable result aggregation across
+		// homogeneous hosts.
+		engineRank, err := engine.GetRank()
+		if err != nil {
+			engine.Debugf("instance %d GetRank: %s", engine.Index(), err.Error())
```

Comment:
Why debug rather than error? Is there a concern that there may be false errors logged?

Reply:
I didn't want to refuse the scan just because the rank could not be read, do you think I should return an error here?

Reply:
I was thinking of logging the error, at least. It should never be a requirement to restart the servers at debug level to learn about a user-relevant problem.

Reply:
Sure, sounds like a good idea. are you okay with me making the change in a subsequent PR?

Reply:
Yes, that's fine.

Reply:
Fixed in https://github.com/daos-stack/daos/pull/13605/files#diff-62229de19ab79684f56f742628359e92e773d74366952556da248193fb6afe1bR221 . I can't log to Error without adding another method to the Engine interface but I think it's fine to return an error in this case because realistically the rank should be assigned if the engine is to be queryable over dRPC at this point.